tests/Doctrine/Tests/ORM/Functional/CustomFunctionsTest.php

```diff
@@ -64,6 +64,31 @@ public function testCustomFunctionOverride()
 
         $this->assertEquals(1, $usersCount);
     }
+
+    public function testCustomFunctionWithinInExpression() : void
+    {
+        $user           = new CmsUser();
+        $user->name     = 'Lo';
+        $user->username = 'Doe';
+
+        $this->_em->persist($user);
+        $this->_em->flush();
+
+        $this->_em->getConfiguration()->addCustomStringFunction(
+            'FOO',
+            function (string $funcName) : NoOp {
+                return new NoOp($funcName);
+            }
+        );
+
+        $query = 'SELECT u FROM Doctrine\Tests\Models\CMS\CmsUser u WHERE u.name IN (FOO(\'Lo\'), \'Lo\', :name)';
+        $users = $this->_em->createQuery($query)
```

Comment:
This test should be changed to an SQL generation test, in my opinion. Persistence logic is not needed here

Reply:
@Ocramius Can you provide me an example where I can look at? I'm not quite sure what you mean here.

Reply:
See https://github.com/doctrine/doctrine2/blob/6a861756176a6bfcfe5c4ecb699e3b08c7582cfb/tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php